pkg/arvo/lib/graph-view.hoon

```diff
@@ -0,0 +1,80 @@
+/-  sur=graph-view
+/+  resource, group-store
+^?
+=<  [sur .]
+=,  sur
+|%
+++  app-to-mark
+  |=  app=app-name
+  ?+  app  ~|("nonexistent app" !!)
+    %chat  %graph-validator-chat
+    %publish  %graph-validator-publish
+  ==
+++  dejs
+  =,  dejs:format
+  |%
+  ++  action
+    |^
+    ^-  $-(json ^action)
+    %-  of
+    :~  create+create
+        delete+delete
+        join+join
+        leave+leave
+        groupify+groupify
+        ::invite+invite
+    ==
+    ::
+    ++  create
+      %-  ot
+      :~  resource+dejs:resource
+          title+so
+          description+so
+          app+app
+          associated+associated
+      ==
+    ::
+    ++  app
+      |=  jon=json
+      ^-  app-name
+      ?>  ?=(%s -.jon)
+      ?+  p.jon  !!
+        %chat     %chat
+        %publish  %publish
+      ==
+    ::
+    ++  leave
+      %-  ot
+      :~  resource+dejs:resource
+          app+app
+      ==
+    ::
+    ++  delete
+      %-  ot
+      :~  resource+dejs:resource
+          app+app
+      ==
+    ::
+    ++  join
+      %-  ot
+      :~  resource+dejs:resource
+          app+app
+          ship+(su ;~(pfix sig fed:ag))
```

Comment:
Why do we need ship? Could we not use the resource's ship?

Reply:
Was more in the interest of maintaining a bijection between the JSON and the actual interface of the app